bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BIntersectionType.java

```diff
@@ -188,6 +202,11 @@ public void setImmutableType(IntersectionType immutableType) {
         this.immutableType = immutableType;
     }
 
+    @Override
+    public Type getReferredType() {
```

Comment:
Isn't this meant for type reference types? Is it correct to override it this way?

Reply:
will have to change this thanx.